File: programmers/level1/약수의 합.py
def solution(n):
    answer = 0
    for i in range(1, int(n ** 0.5) + 1):
        if n % i == 0:
            answer += i
            if i != n // i:
                answer += n // i
    return answer


print(solution(12))
print(solution(5))
print(solution(0))
print(solution(1))
print(solution(2))
print(solution(4))
print(solution(5))
print(solution(6))

File: programmers/level1/약수의 합/약수의 합.go
package

func solution(n int) (sum int) {
	if n == 0 || n == 1 {
		return n
	}

	divisors := map[int]struct{}{}

	for i := 1; i < n/2+1; i++ {
		if n%i == 0 {
			divisors[i] = struct{}{}
			divisors[n/i] = struct{}{}
		}
	}

	for divisor := range divisors {
		sum += divisor
	}

	return
}
